external/kinesis-asl-assembly/pom.xml

```diff
@@ -141,6 +141,21 @@
   <outputDirectory>target/scala-${scala.binary.version}/classes</outputDirectory>
   <testOutputDirectory>target/scala-${scala.binary.version}/test-classes</testOutputDirectory>
   <plugins>
+    <!-- SPARK-17418: prevent the kinesis-asl-assembly from being published to Maven -->
+    <plugin>
+      <groupId>org.apache.maven.plugins</groupId>
+      <artifactId>maven-deploy-plugin</artifactId>
+      <configuration>
+        <skip>true</skip>
+      </configuration>
+    </plugin>
+    <plugin>
+      <groupId>org.apache.maven.plugins</groupId>
+      <artifactId>maven-install-plugin</artifactId>
```

Comment:
We need to do this in order to prevent this artifact from being published to a release staging repository, since the current release publishing scripts work by first installing the components to be published into a fresh temporary Maven repository and then searching that repository for artifacts, signing them, and uploading them to the staging repository using `curl`. See https://github.com/apache/spark/blob/7e418e99cff4cf512ab2a9fa74221c4655048c8d/dev/create-release/release-build.sh#L294